lints/lint_rsa_signature_algo_parameter_not_null.go

```diff
@@ -0,0 +1,118 @@
+
+func (l *rsaEncryptionParamNotNULL) Initialize() error {
+	return nil
+}
+
+func (l *rsaEncryptionParamNotNULL) CheckApplies(c *x509.Certificate) bool {
+	// explicitly check for util.OidRSAEncryption, as RSA-PSS or RSA-OAEP certificates might be classified with c.PublicKeyAlgorithm = RSA
+	return c.PublicKeyAlgorithmOID.Equal(util.OidRSAEncryption)
+}
+
+func (l *rsaEncryptionParamNotNULL) Execute(c *x509.Certificate) *LintResult {
+	// Try byte comparison for Algorithm
+	// Determine offset (SEQUENCE + length) first. Notably 1024 encodes length in 2 bytes, 2048 and up in 3 bytes
+	offset := getOffset(c.RawSubjectPublicKeyInfo)
+	if len(c.RawSubjectPublicKeyInfo) <= len(expectedSPKIAlgoBytes)+offset {
+		return &LintResult{Status: Fatal, Details: "error malformed pkixPublicKey"}
+	}
+
+	if bytes.Compare(c.RawSubjectPublicKeyInfo[offset:len(expectedSPKIAlgoBytes)+offset], expectedSPKIAlgoBytes) == 0 {
+		return &LintResult{Status: Pass}
+	}
+
+	input := cryptobyte.String(c.RawSubjectPublicKeyInfo)
+
+	var publicKeyInfo cryptobyte.String
+	if !input.ReadASN1(&publicKeyInfo, asn1_cryptobyte.SEQUENCE) {
+		return &LintResult{Status: Fatal, Details: "error reading pkixPublicKey"}
+	}
+
+	var algorithm cryptobyte.String
+	if !publicKeyInfo.ReadASN1(&algorithm, asn1_cryptobyte.SEQUENCE) {
+		return &LintResult{Status: Fatal, Details: "error reading pkixPublicKey algorithm"}
+	}
+
+	if !algorithm.SkipASN1(asn1_cryptobyte.OBJECT_IDENTIFIER) {
+		return &LintResult{Status: Fatal, Details: "error reading pkixPublicKey algorithm OID"}
+	}
+
+	if algorithm.Empty() {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier missing required NULL parameter"}
+	}
+
+	var nullValue cryptobyte.String
+	if !algorithm.ReadASN1(&nullValue, asn1_cryptobyte.NULL) {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier with non-NULL parameter"}
+	}
+
+	if len(nullValue) != 0 {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier with NULL parameter containing trailing data"}
+	}
+
+	// ensure algorithm is empty and no trailing data is present
+	if !algorithm.Empty() {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier with trailing data"}
+	}
+
+	return &LintResult{Status: Fatal, Details: "certificate rsa algorithm identifier appears correct, but didn't match byte-wise comparison"}
+}
+
+func init() {
+	RegisterLint(&Lint{
+		Name:          "e_rsa_encryption_parameter_not_null",
+		Description:   "RSA: Encoded algorithm identifier MUST have NULL parameters",
+		Citation:      "RFC 4055, Section 1.2",
+		Source:        RFC5280, // RFC4055 is referenced in RFC5280, Section 1
+		EffectiveDate: util.RFC5280Date,
+		Lint:          &rsaEncryptionParamNotNULL{},
+	})
+}
+
+func getOffset(asn1Sequence []byte) int {
```

Comment:
Can the name of this function be updated to be a little bit more informative?

If I'm understanding correctly this is computing the offset to the start of the algorithm object identifier `SEQUENCE` byte offset in the `RawSubjectPublicKeyInfo` `SEQUENCE`. If that's accurate maybe a clearer name would be `getOIDOffset` ?

Reply:
@tadukurow Based on discussion in [the other feedback thread](https://github.com/zmap/zlint/pull/286#discussion_r289924518) I think you can replace `getOffset` entirely by using `cryptobyte` to unpack your way into the `RawSubjectPublicKeyInfo` to the right offset for the comparison. Most of my other feedback will be resolved by switching this code out.

I think that will be more readable and remove the DER-specific details in `getOffset` that are unrelated to the linting. 

Thanks!